### PR TITLE
Remove DataID attribute access

### LINKED ISSUE #2395: DataID.__getattr__ calls nonexisting method on parent class
**Describe the bug**

In `DataID.__getattr__`, there is an implementation of attribute access for DataIDs for ID keys, for backward compatibility reasons.  Any other attribute access is delegated to `super().__getattr__`.  However, the parent class `dict` does not define a `__getattr__` method, so this results in `AttributeError`.

Now, it _should_ result in `AttributeError`, but due to the missing attribute being requested, not due to `__getattr__` being missing on the parent class.

**To Reproduce**

```python
from satpy.dataset.dataid import DataID
did = DataID({})
did.unknown
```

**Expected behavior**

I expect an `AttributeError` because the indicated attribute is unknown:

```python-traceback
Traceback (most recent call last):
  File "/data/gholl/checkouts/protocode/mwe/did-attributeerror.py", line 3, in <module>
    did.unknown
AttributeError: 'DataID' object has no attribute 'unknown'
```

**Actual results**

Instead I'm getting an `AttributeError` because the super object has no attribute `__getattr__`:

```python-traceback
Traceback (most recent call last):
  File "/data/gholl/checkouts/protocode/mwe/did-attributeerror.py", line 3, in <module>
    did.unknown
  File "/data/gholl/checkouts/satpy/satpy/dataset/dataid.py", line 410, in __getattr__
    return super().__getattr__(key)
           ^^^^^^^^^^^^^^^^^^^
AttributeError: 'super' object has no attribute '__getattr__'. Did you mean: '__setattr__'?
```

**Environment Info:**
 - Satpy Version: main

### PR DESCRIPTION
**Pull request by gerritholl**
Remove the `__getattr__` implementation on DataID, that was implementing a deprecated way to access information from DataID objects.  From now on, only key access is possible.

 - [x] Closes #2395
 - [x] Tests adapted

### CODE REVIEW

File: satpy/dataset/dataid.py [hunk 1]
```diff
@@ -407,7 +407,7 @@ def __getattr__(self, key):
                           stacklevel=2)
             return self[key]
         else:
-            return super().__getattr__(key)
+            raise AttributeError(f"'{type(self).__name__:s}' object has no attribute '{key:s}'")
```
[the remark this answers is not in the document]
Comment (gerritholl) [reply satpy/dataset/dataid.py:410]: `__getattr__` only gets called for missing attributes, so this doesn't stop from accessing methods from the dict class.

I would be OK with removing `__getattr__` completely.

[the remark this answers is not in the document]
Comment (djhoese) [reply satpy/dataset/dataid.py:410]: @mraspaud thoughts?

[the remark this answers is not in the document]
Comment (mraspaud) [reply satpy/dataset/dataid.py:410]: there is no code better than no code. Just remove it :)

[the remark this answers is not in the document]
Comment (gerritholl) [reply satpy/dataset/dataid.py:410]: Easy, only 564 test failures when I do that! ☺

[the remark this answers is not in the document]
Comment (djhoese) [reply satpy/dataset/dataid.py]: My guess is SIFT (CC @strandgren @ameraner) and maybe my Polar2Grid and Geo2Grid stuff.

[the remark this answers is not in the document]
Comment (ameraner) [reply satpy/dataset/dataid.py]: Yes indeed, we need to check (I spotted one already some time ago, see [this comment in the EUM Gitlab](https://gitlab.eumetsat.int/webservices/mtg-sift/-/issues/138#note_114648)). We even have some usages of `_satpy_id` in the code...

[the remark this answers is not in the document]
Comment (mraspaud) [reply satpy/dataset/dataid.py]: @ameraner I can't access the EUM gitlab apparently, I need an account.

**Pull request merged**